$NetBSD: patch-aa,v 1.9 2026/10/02 13:01:27 gdt Exp $

Add stanza for modern NetBSD.  \todo Send upstream.

Adjust Makefile for pkgsrc norms:

  - Use PREFIX
  - Use LDFLAGS (relro)
  - Comment out FreeBSD
  - Uncomment the "fail if not configured" line
  - Change installation to use INSTALL_PROGRAM
    \todo Evaluate PROGRAM vs SCRIPT; this is carried from before.

--- Makefile.orig	2023-02-15 11:25:27.000000000 +0000
+++ Makefile
@@ -48,12 +48,13 @@
 
 
 CC?=cc
-CFLAGS?=-O2
+# Use CFLAGS form the environment.
 
 #paths:
 #some peple like /usr/local/sbin instead of /usr/local/etc
-BINDIR=/usr/local/bin
-ETCDIR=/usr/local/etc
+SBINDIR=${PREFIX}/sbin
+BINDIR=${PREFIX}/bin
+ETCDIR=${PREFIX}/etc
 PRINTCMD=enscript -Gr2
 # you only need RSAREF for ESM
 # these paths WILL have to be changed...
@@ -188,6 +189,13 @@ RINCLUDES=/usr/mab/rsaref/source
 #COMPAT=-lcompat
 #RPCOPTS=-b
 
+#* Use these for NetBSD 9
+# For mounting, use
+#      mount -o intr,udp,-2 127.0.0.1:/null /crypt
+#CFLAGS=$(COPT) -DBSD44 -DANYPORT -DCFS_PORT=2049 -DSHORTLINKS -I$(RINCLUDES)
+#COMPAT=-lcompat
+#RPCOPTS=-b
+
 ## Solaris 2.3 / SUNOS 5.x
 #CFLAGS=$(COPT) -DSOLARIS2X -DPORTMAP -I$(RINCLUDES) -DPTMX
 #LIBS=-lsocket -lnsl
@@ -198,17 +206,17 @@ RINCLUDES=/usr/mab/rsaref/source
 #CFLAGS=$(COPT) -posix -D_BSD -DANYPORT -I$(RINCLUDES)
 
 ## use these for FreeBSD
-CFLAGS+=-DBSD44 -DANYPORT -DSHORTLINKS
-LIBS=-lrpcsvc
-COMPAT=-lcompat
-RPCOPTS=
+#CFLAGS+=-DBSD44 -DANYPORT -DSHORTLINKS
+#LIBS=-lrpcsvc
+#COMPAT=-lcompat
+#RPCOPTS=
 
 
 #==========================================================================
 # CONFIGURE: one last thing
 #==========================================================================
 # finally, comment out the next line:
-CC=you_forgot_to_edit_the_makefile
+#CC=you_forgot_to_edit_the_makefile
 
 # now you're done with local configuration.
 
@@ -244,41 +252,41 @@ cfs: cfsd cattach cmkdir cdetach cname c
 	@echo
 
 cfsd: $(OBJS)
-	$(CC) $(OBJS) $(LIBS) -o cfsd
+	$(CC) $(OBJS) ${LDFLAGS} $(LIBS) -o cfsd
 
 cattach: cattach.o admproto_clnt.o admproto_xdr.o getpass.o cfs_des.o \
   cfs_cipher.o cfs_bfenc.o cfs_bfsk.o adm.o mcg.o mcgsbox.o shs.o safer.o
 	$(CC) cattach.o admproto_clnt.o admproto_xdr.o cfs_des.o \
 	   cfs_bfenc.o cfs_bfsk.o cfs_cipher.o getpass.o adm.o mcg.o \
-	   mcgsbox.o shs.o safer.o $(COMPAT) $(LIBS) -o cattach
+	   mcgsbox.o shs.o safer.o $(LDFLAGS) $(COMPAT) $(LIBS) -o cattach
 
 cdetach: cdetach.o admproto_clnt.o admproto_xdr.o adm.o
 	$(CC) cdetach.o adm.o admproto_clnt.o admproto_xdr.o \
-	   $(LIBS) -o cdetach
+	   $(LDFLAGS) $(LIBS) -o cdetach
 
 cmkdir: getpass.o adm.o cfs_des.o cfs_cipher.o cmkdir.o mcg.o \
    mcgsbox.o cfs_bfenc.o cfs_bfsk.o safer.o shs.o truerand.o
 	$(CC) cmkdir.o cfs_des.o cfs_cipher.o getpass.o adm.o mcg.o \
 	   cfs_bfenc.o cfs_bfsk.o mcgsbox.o safer.o shs.o truerand.o \
-	   $(COMPAT) -o cmkdir
+	   $(LDFLAGS) $(COMPAT) -o cmkdir
 
 cpasswd: getpass.o cfs_des.o cfs_cipher.o cpasswd.o mcg.o \
    mcgsbox.o cfs_bfenc.o cfs_bfsk.o safer.o shs.o truerand.o
 	$(CC) cpasswd.o cfs_des.o cfs_cipher.o getpass.o mcg.o \
 	   mcgsbox.o cfs_bfenc.o cfs_bfsk.o safer.o shs.o \
-	   truerand.o  $(COMPAT) -o cpasswd
+	   truerand.o  $(LDFLAGS) $(COMPAT) -o cpasswd
 
 cname: cname.o getpass.o cfs_des.o cfs_cipher.o cfs_adm.o cfs_fh.o \
    cfs_bfenc.o cfs_bfsk.o cfs_nfs.o mcg.o mcgsbox.o safer.o shs.o
 	$(CC) cname.o getpass.o cfs_des.o cfs_cipher.o cfs_adm.o cfs_fh.o \
 	   cfs_nfs.o mcg.o mcgsbox.o cfs_bfenc.o cfs_bfsk.o safer.o \
-	   shs.o $(LIBS) $(COMPAT) -o cname
+	   shs.o $(LDFLAGS) $(LIBS) $(COMPAT) -o cname
 
 ccat: ccat.o getpass.o cfs_des.o cfs_cipher.o cfs_adm.o cfs_fh.o cfs_nfs.o \
    mcg.o mcgsbox.o shs.o safer.o
 	$(CC) ccat.o getpass.o cfs_des.o cfs_cipher.o cfs_adm.o cfs_fh.o \
 	   cfs_bfenc.o cfs_bfsk.o cfs_nfs.o mcg.o mcgsbox.o shs.o \
-	   safer.o $(LIBS) $(COMPAT) -o ccat
+	   safer.o $(LDFLAGS) $(LIBS) $(COMPAT) -o ccat
 
 $(OBJS): nfsproto.h admproto.h cfs.h mcg.h safer.h shs.h \
    cfs_bf.h cfs_bfl.h cfs_bfp.h
@@ -344,10 +352,11 @@ printout: $(SRCS) cfs.h mcg.h safer.h ad
 	$(PRINTCMD) $(SRCS) cfs.h mcg.h safer.h admproto.h nfsproto.h
 
 install_cfs: cfsd cattach cdetach cmkdir
-	install -m 0755 -c -o root cfsd $(ETCDIR)
-	install -m 0755 -c -o root cattach cdetach cmkdir cpasswd cfssh \
-                cname ccat cmkkey $(BINDIR)
-#	install -m 0755 i o $(BINDIR)
+	${BSD_INSTALL_PROGRAM} cfsd ${DESTDIR}$(SBINDIR)
+	${BSD_INSTALL_SCRIPT} cmkkey cfssh ${DESTDIR}$(BINDIR)
+	${BSD_INSTALL_SCRIPT} cattach cdetach cmkdir cpasswd \
+                cname ccat ${DESTDIR}$(BINDIR)
+#	${BSD_INSTALL_PROGRAM} i o ${DESTDIR}$(BINDIR)
 	@echo "Kill any running cfsd prior to restarting."
 	@echo "See the README file for more information."
 	@echo "Don't forget to install the man pages (*.[18])."
